src/components/movies/Movies.js

```diff
@@ -1,41 +1,44 @@
 import MovieCard from "./card/MovieCard";
 import React, {useEffect} from "react";
 import classes from "./Movies.module.css";
-import PropTypes from "prop-types";
 import {filterMovies} from "../../redux/actions";
 import {connect} from "react-redux";
+import {useNavigate, useParams, useSearchParams} from "react-router-dom";
 
 function mapStateToProps(state) {
   const {movies} = state
   return {
     movies: movies?.data,
-    currentGenre: state.currentGenre,
-    currentParameter: state.currentParameter,
-    currentOrder: state.currentOrder,
+    currentOrder: state.currentOrder
   }
 }
 
 const mapDispatchToProps = dispatch => ({
-  filterMovies: () => dispatch(filterMovies())
+  filterMovies: (sortBy, genre, search) => dispatch(filterMovies(sortBy, genre, search))
 })
 
 function Movies(props) {
+  const navigate = useNavigate();
+  const {searchQuery} = useParams();
+  let [searchParams] = useSearchParams();
+
+  const showDescription = (id) => {
+    searchParams.set("movie", id);
+    navigate(`/search${searchQuery ? "/" + searchQuery : ""}?${searchParams.toString()}`)
+  }
+
   useEffect(() => {
-    props.filterMovies()
-  }, [props.currentGenre, props.currentParameter, props.currentOrder])
+    props.filterMovies(searchParams.get("sortBy"), searchParams.get("genre"), searchQuery)
+  }, [searchParams.get("sortBy"), searchParams.get("genre"), searchQuery, props.currentOrder])
 
   return (
     <div className={classes.movies}>
       {
         props.movies.map(value => <MovieCard key={props.movies.indexOf(value)} movie={value}
-                                             showDescription={props.showDescription}/>)
+                                             showDescription={showDescription}/>)
```

Comment:
Лучший подход не перегружать простые компоненты лишней логикой
```suggestion
                                             showDescription={() => showDescription(value.id)}/>)
```

Reply:
Done